HW1Review

tequillajazz opened the pull request.

tequillajazz committed:
HW1Review

numash commented:
Hi Andrii! Great job on this task, I like your code :)

Could you please fix several issues?

1. There's a misleading error message for 0 input

![image](https://user-images.githubusercontent.com/9382934/109999349-ce60bc80-7d1a-11eb-9866-6ef4a08d157e.png)

2. I couldn't find any constants in your code. Could you please use them? (maybe consider adding them to a separate file)
3. As I can see, there's no restriction for the initial input to be < than 5 in the HW task.
4. There's an error if I enter any character after the "Press any key..." message
![image](https://user-images.githubusercontent.com/9382934/110001518-1d0f5600-7d1d-11eb-928b-716cd1454cf8.png)


Questions:
1. Could you please explain why in class Helper the RandomNumberBetween method is static, but the GetValueFromConsole is not?
2. Why do you need this row - AreaCalculator areaCalculator = new AreaCalculator();

Suggestions:
1. You can use auto property for Radius and SquareArea properties.
2. You can also create Area property instead of the GetArea() method.

tequillajazz committed:
HWAQA1 - Review fixes

tequillajazz commented:
> Hi Andrii! Great job on this task, I like your code :)
> 
> Could you please fix several issues?
> 
> 1. There's a misleading error message for 0 input
> 
> ![image](https://user-images.githubusercontent.com/9382934/109999349-ce60bc80-7d1a-11eb-9866-6ef4a08d157e.png)
> 
> 1. I couldn't find any constants in your code. Could you please use them? (maybe consider adding them to a separate file)
> 2. As I can see, there's no restriction for the initial input to be < than 5 in the HW task.
> 3. There's an error if I enter any character after the "Press any key..." message
>    ![image](https://user-images.githubusercontent.com/9382934/110001518-1d0f5600-7d1d-11eb-928b-716cd1454cf8.png)
> 
> Questions:
> 
> 1. Could you please explain why in class Helper the RandomNumberBetween method is static, but the GetValueFromConsole is not?
> 2. Why do you need this row - AreaCalculator areaCalculator = new AreaCalculator();
> 
> Suggestions:
> 
> 1. You can use auto property for Radius and SquareArea properties.
> 2. You can also create Area property instead of the GetArea() method.

1) Fixed
2) Used Math.PI instead of constants in the code
3) That was my chooise=)
4) Fixed

Q1 Fixed
Q2 Fixed

numash commented:
Hi Andrii, 
I've reviewed your fixes, everything looks good.

Using constants is one of the tasks for HW1. You can use them not only for math constants but also e.g. for the number of tries.
Let's do the following: this task can be accepted.
But in HW2 you'll

- use at least one constant
- instead of the big structures:

private double _squareArea;
		public double SquareArea
		{
			get { return _squareArea; }
			set
			{
				_squareArea = value;
			}
		}

use autoproerty.
I hope, it works for you
